app/src/main/java/com/keylesspalace/tusky/BaseActivity.java

```diff
@@ -222,4 +235,37 @@ public void showAccountChooserDialog(CharSequence dialogTitle, boolean showActiv
             .setAdapter(adapter, (dialogInterface, index) -> listener.onAccountSelected(accounts.get(index)))
             .show();
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode != REQUESTER_NONE && requesters.containsKey(requestCode)) {
+            requesters.remove(requestCode).onRequestPermissionsResult(permissions, grantResults);
+        }
+    }
+
+    public boolean requestPermissions(String[] permissions, int minimumBuildVersion, @Nullable PermissionRequester requester) {
```

Comment:
I would rather accept lambda or returned `Single` here. Can be such a fluent API:
```kotlin
requestPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) { givenPermissions ->
   if (givenPermissions.isNotEmpty()) {
       downloadFile()
   |
}
```

Reply:
This way, it can directly chain the callback from `ActivityCompat.requestPermissions`

Reply:
...but I did update it so that the callsite as lambdaable as java allows...